StorageUtility/DeviceInterface.h

```diff
@@ -0,0 +1,46 @@
+/*
+<License>
+Copyright 2015 Virtium Technology
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http ://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+</License>
+*/
+#ifndef __DeviceInterface_h__
+#define __DeviceInterface_h__
+#pragma once
+
+#include <map>
+#include <memory>
+#include <vector>
+
+#include "DeviceDataType.h"
+
+namespace vtStor
+{
+
+class cDeviceInterface
+{
+protected:
+    cDeviceInterface();
+    virtual ~cDeviceInterface();
+
+public:
+    virtual const std::map<eDeviceDataType, void*>& Data() = 0;
```

Comment:
1. std::map should be std::unordered_map.
2. Data() should take in a reference of a container instead of returning a container. As Lan said, "making copies is not cheap."